### ISSUE TITLE
issue with conditionCounter -- only incrementing

### ISSUE THREAD

**Issue by prakash-indorkar** [jdgamble555/adv-firestore-functions#12]
Hi  this "adv-firestore-functions@1.7.8" version solved the first problem of conditionCounter modifying count value when the record is created. 

Now there is another issue I found, ( for my case this is not an issue, as I do not allow user to change the value of isNice again). 
So if the user changes his mind and if there is an option to undo isNice then the count value remains the same. And if again change of same value, it increments the counter.

for example:

1 record .. isNice = false
2 record .. isNice = false

isNiceCount = 0

when I update them (one by one)
1 record .. isNice = true
2 record .. isNice = true

isNiceCount = 2

3rd record with an initial value false

1 record .. isNice = true
2 record .. isNice = true
3 record .. isNice = false

isNiceCount = 2

3rd record with a change value true

1 record .. isNice = true
2 record .. isNice = true
3 record .. isNice = true

isNiceCount = 3

3rd record with a change value false again

1 record .. isNice = true
2 record .. isNice = true
3 record .. isNice = false

isNiceCount = 3 -------> remains the same

3rd record with a change value true again

1 record .. isNice = true
2 record .. isNice = true
3 record .. isNice = true

isNiceCount = 4 -------> incremented value.

the code looks like this 

```javascript
exports.niceCount = functions.firestore.document('/users/{docId}')
    .onWrite(async (change, context) => {
        await conditionCounter(change, context, 'isNice', '==', true);
        return null;
    });
```

**Comment by prakash-indorkar** [comment]
I think this can be solved by having a boolean parameter .. say considerDefault value. If true then it will consider the default value to update the counter and false will not consider it for the count. ( So your previous version could work, where the count value was updating on any change).

**Comment by jdgamble555** [comment]
I did that, just simple mistake.  Always pass in **change** variable to **createDoc**!

Try it now, should work.  Version **1.80**

**Comment by prakash-indorkar** [comment]
Hi .. the issue is still there. this time when you change to the default value, the count becomes 0, despite having other records with value true.

While you fix the issue I Just had a question - I often get this warning whenever cloud functions are invoked
`(node:17900) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 log listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit`

is this something related to your package? I googled it and found that it was fixed with an upgrade version of firebase-admin, but I already have the upgraded version, I do not know why this warning is still coming.

**Comment by jdgamble555** [comment]
Still working on it, may not get to it until the weekend, but will get it fixed.  As far as your other question, not sure.  Make sure your node is up to date as well.

**Comment by jdgamble555** [comment]
Should be fixed now with 1.8.9... let me know if there are any problems...

**Comment by prakash-indorkar** [comment]
Thanks man!.. This one fixed now.
